### ISSUE TITLE
Backup Virtual Servers - Readability improvement in the script output text

### ISSUE THREAD

**Issue by shoulders** [virtualmin/virtualmin-gpl#918]
| SYSTEM INFORMATION ||
|----------|----------|
|  OS type and version |  Ubuntu Linux 22.04.5             |
|  Usermin version     |  2.102                            |
|  Virtualmin version  |  7.20.2 Pro                       |
|  Theme version       |  21.20.7                          |
|  Apache version      |  2.4.52                           |
|  Package updates     |  2 package updates are available  |

### Background

When you backup virtual servers you get an output on screen telling you what is going on.

### The issue

1. Although the backup all works I feel that the output text could be a little clearer so the end users can see what is going on easier, in particular the separation between the virtual servers.
2. When you add a virtualmin system config this section does not have a header
3. `Creating final backup archive` section should be made clearer
4. When you add a virtualmin system config the summary at the end does not indicate the config has been backed up

![image](https://github.com/user-attachments/assets/53f491b1-6672-47de-9fb9-5acb70c48fb2)

### Proposed Solutions

1. Bold the `Creating backup for virtual server example.com ..`
2. Add a heading to the `Virtualmin system config` section in bold.
3. Creating final backup archive should be bolded
4. Add some text to the summary at the bottom indicating the Virtualmin system config has been added.
5. (optional) further separate the summary section with a bolded header

![image](https://github.com/user-attachments/assets/cedcfc0b-6495-4ce1-a3f5-4a35e006f4ae)

**Comment by iliajie** [comment]
Hello,

Thanks for your suggestions!

> 2. Add a heading to the `Virtualmin system config` section in bold.

Thanks for pointing this out!

Yes, this is a bug and should be fixed! We need to wait for Jamie's comment on this though!

> 1. Bold the `Creating backup for virtual server example.com ..`
> 3. Creating final backup archive should be bolded
> 4. Add some text to the summary at the bottom indicating the Virtualmin system config has been added.
> 5. (optional) further separate the summary section with a bolded header

Well, this isn't something common for Virtualmin, and I personally don't like how it looks.

**Comment by jcameron** [comment]
Thanks, I've checked in a fix for that missing heading.

As for the other suggestions, I don't really like bolding as it would be inconsistent with the rest of Virtualmin.

Also, doesn't the summary already say that Virtualmin settings were backed up?

**Comment by shoulders** [comment]
> Thanks, I've checked in a fix for that missing heading.

Brilliant + thanks

> As for the other suggestions, I don't really like bolding as it would be inconsistent with the rest of Virtualmin.

Fair enough

> Also, doesn't the summary already say that Virtualmin settings were backed up?

- I just ran backing up only the system settings and you are right, but it is not clear. It should be on a separate line and also have the word `system` in it.
- I assumed this was somethin to do with owner accounts because you can have less of those than virtual servers

![image](https://github.com/user-attachments/assets/0587f11a-d618-41a6-a2ed-eba2f7f9da83)

### additional

Personally I think the summary line that is there is not very clear.

- If a server backups up successfully, Can they still have errors? perhaps you don't need to add this unless there is errors?
- 10 servers --> 10 Virtual servers
- I added the word `system` to the virtualmin system configuration line so people know it is referring to the system configs and not account owner configs.
- It is important to put both these on their own line because there are different and might not always be present.
- I changes the order to: Virtual Servers, System
- are configurations ans settings not the same thing or, configurations are made up of settings? So I removed the word `settings`.

I think it would be better if it reads like:

> 10 Virtual servers backed up successfully.
> 8 Virtualmin system configurations backed up successfully.

**Comment by iliajie** [comment]
> I think it would be better if it reads like:
> 
> > 10 Virtual servers backed up successfully.
> > 8 Virtualmin system configurations backed up successfully.

Jamie, why were 10 virtual servers backed up successfully, but only 8 Virtualmin configuration settings?

**Comment by shoulders** [comment]
> Jamie, why were 10 virtual servers backed up successfully, but only 8 Virtualmin configuration settings?

Exactly why I was confused. the `8 Virtualmin configuration settings` refer to the 8 areas of virtualmin system configs that are backed up. However there are more than 8.

![image](https://github.com/user-attachments/assets/7f8184fb-a3a1-468d-b7aa-776208e97471)

**Comment by jcameron** [comment]
>. Exactly why I was confused. the 8 Virtualmin configuration settings refer to the 8 areas of virtualmin system configs that are backed up. However there are more than 8.

Ok that's a bug which I've just checked in a fix for

**Comment by shoulders** [comment]
@jcameron I see you
- have added a title for the virtualmin system config section
- you have corrected the virtualmin system config section count

Thanks

**Some points still to be addressed (on item 4)** 

- If a server backups up successfully, Can they still have errors? perhaps you don't need to add this unless there is errors?
- 10 servers --> 10 Virtual servers
  - They are not servers, they are named as virtual server througout Virtualmin so this should be consistent.
- add the word `system` into the `8 Virtualmin configurations backed up successfully.` summary line
  - I added the word system to the virtualmin system configuration line so people know it is referring to the system configs and not account owner configs.
- put both the success notifications on separate lines.
  - It is important to put both these on their own line because they are different and might not always be present.
- I set the order for the lines: Virtual Servers, System
  - This reads the best
- Are configurations and settings not the same thing or, configurations are made up of settings?
  - I removed the word `settings` as I think it reads better.

This will give the following example and I think it reads better and more logical with these small changes:

> 10 Virtual servers backed up successfully.
> 8 Virtualmin system configurations backed up successfully.

**Comment by jcameron** [comment]
Thanks, I'll implement these suggestions!

**Comment by shoulders** [comment]
After reading the notes there are 2 things missing to completed this issue and I will outline them below:

### Current Backup Output

<img width="654" height="650" alt="Image" src="https://github.com/user-attachments/assets/6f9c34ec-4ce7-4f86-a657-1fd98fb51e04" />

### New Backup Output

<img width="574" height="357" alt="Image" src="https://github.com/user-attachments/assets/22408095-bda5-426a-bfab-785a6c0480b8" />

- **(A)** Add a Summary title
  - This makes the sections consistent with each other. Current the `backed up` statements are not in a section and effectively in the root 😄 
- **(B)** Add a `Virtualmin Settings backed up sucessfully` statement in the summary section.
  - Again this is about consistency. Some might say, it says they were backed up just above, then what is the point in the summary then.

### Adidtional
I know some of my earlier points were about bolding the titles, however these should be `marked as not planned` and then this issue can be closed if **A** and **B** get approved and done.

**Comment by iliaross** [comment]
So, are you suggesting moving the last two lines under the summary "section"?

**Comment by shoulders** [comment]
yes, and adding an extra line when virtualmin settings are added.

**Comment by shoulders** [comment]
and adding a summary section/title

**Comment by shoulders** [comment]
i don't thinks so. I think it is important to see in the summary what has been added and while the virtualmin settings have to be here then they deserve they own entry in the summary.

A summary is just that, not a curated list 😄

**Comment by shoulders** [comment]
I see what you mean

For some reason I thought it ment:
> the first line is the number of virtualservers in total, then the next one is the number of user accounts because some parent accounts can have many sub-servers.

I was looking for the term `Virtualmin settings` and not `Virtualmin configurations`. Perhaps change one so both options have the same name.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
